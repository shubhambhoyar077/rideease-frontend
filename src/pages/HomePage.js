import React, { useState } from 'react';
import Cars from '../components/Cars';
import '../styles/HomePage.css';

const HomePage = () => {
  const [carouselIndex, setCarouselIndex] = useState(0);

  const scrollRight = () => {
    setCarouselIndex((prevIndex) => prevIndex + 1);
  };

  const scrollLeft = () => {
    setCarouselIndex((prevIndex) => prevIndex - 1);
  };

  return (
    <section className="home-page">
      <h1>LATEST MODELS</h1>
      <p className="home-page-text">Please select a Car Model</p>
      <div className="carousel-container">
        <Cars startIndex={carouselIndex} endIndex={carouselIndex + 3} />
        <div className="carousel-buttons">
          <button
            type="button"
            className="btn-cr2"
            onClick={scrollLeft}
            disabled={carouselIndex === 0}
          >
            {'<'}
          </button>
          <button
            type="button"
            className="btn-cr"
            onClick={scrollRight}
            disabled={carouselIndex >= 3}
          >
            {'>'}
          </button>
        </div>
      </div>
    </section>
  );
};

export default HomePage;
