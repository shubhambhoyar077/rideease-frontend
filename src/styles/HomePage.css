* {
  padding: 0;
  margin: 0;
  list-style: none;
  text-decoration: none;
}

.home-page {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.home-page-text {
  color: #a2a4a4;
  font-weight: bold;
}

.dots {
  color: #d3d4d4;
  font-size: 24px;
  margin-bottom: 30px;
}

.cars-container {
  display: flex;
  gap: 100px;
}

.cars-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
}

.cars-image {
  width: 15rem;
  height: 15rem;
  border: 5px #00b7ff solid;
  border-radius: 50%;
}

.cars-card-details {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.dots2 {
  color: #d3d4d4;
  font-size: 24px;
}

.car-details {
  font-size: 14px;
  color: #a2a4a4;
}
