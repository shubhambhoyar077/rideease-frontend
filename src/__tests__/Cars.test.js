import React from 'react';
import 'matchmedia-polyfill';
import 'matchmedia-polyfill/matchMedia.addListener';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import '@testing-library/jest-dom';
import HomePage from '../pages/HomePage';

const mockStore = configureMockStore([]);

describe('HomePage and Cars Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      cars: {
        cars: [
          {
            id: 1,
            name: 'Test Car',
            details: 'test details',
            image:
              'https://www.autocar.co.uk/sites/autocar.co.uk/files/images/car-reviews/first-drives/legacy/99-best-luxury-cars-2023-bmw-i7-lead.jpg',
          },
        ],
      },
    });
    store.dispatch = jest.fn();
  });

  it('should render the cars from store', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <HomePage />
        </Provider>
      </BrowserRouter>,
    );

    expect(screen.getByText('LATEST MODELS')).toBeInTheDocument();
    expect(screen.getByText('Please select a Car Model')).toBeInTheDocument();
    expect(screen.getByText('Test Car')).toBeInTheDocument();
    expect(screen.getByText('test details')).toBeInTheDocument();
  });
});
