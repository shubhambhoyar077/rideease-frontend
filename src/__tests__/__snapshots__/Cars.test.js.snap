// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cars Component should render without crashing 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
